source/distributed/tria.cc

```diff
@@ -2165,6 +2165,15 @@ namespace parallel
       n_attached_datas(0),
       n_attached_deserialize(0)
     {
+      // for multigrid, we need limit_level_difference_at_vertices
+      // to make sure the transfer operators only need to consider two levels.
+      if (settings & construct_multigrid_hierarchy)
+        {
+          const typename dealii::Triangulation<dim,spacedim>::MeshSmoothing mg_smooth_grid
+            = static_cast<typename dealii::Triangulation<dim,spacedim>::MeshSmoothing>
+              (smooth_grid | Triangulation<dim,spacedim>::limit_level_difference_at_vertices);
+          this->set_mesh_smoothing(mg_smooth_grid);
```

Comment:
I first preferred readability over modifying smooth_grid after setting it in the initializer list.
But if both of you agree, I don't mind to change this.